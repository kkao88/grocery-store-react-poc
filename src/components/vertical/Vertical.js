import React, { Component } from 'react';
import { Link } from 'react-router-dom';

const _ = require('lodash');

class Vertical extends Component {
    constructor(props) {
        super(props);
    }

    selectVertical = function(input) {
        this.props.onChange(input);
    };

    isSelectedCategory = function(){
      return this.props.paramCategory == this.props.category.id;
    };

    listSubcategories = function(){
        let that = this;
        if (this.props.category && this.isSelectedCategory() && this.props.category.categories){
            return this.props.category.categories.map(function(subcat, i){
                return (
                    <li key={subcat.id}>
                        <Link to={`/category/${that.props.category.id}/${subcat.id}`} onClick={that.props.onClearSearch}>
                            <button className="btn">{subcat.name}</button>
                        </Link>
                    </li>
                );
            });
        }
        else {
            return '';
        }
    };

    render() {
        return (
            <div>
                <span>{this.isSelectedCategory() ? '->' : ''}</span>
                <Link to={`/category/${this.props.category.id}`} onClick={this.props.onClearSearch}>
                    <button className="btn">{this.props.category.name}</button>
                </Link>
                <ul>{this.listSubcategories()}</ul>
            </div>
        );
    }
}

export default Vertical;