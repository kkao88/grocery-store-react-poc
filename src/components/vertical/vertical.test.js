import React from 'react';
import {shallow} from 'enzyme';

import Vertical from './Vertical.js';

import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('Vertical renders correctly', () => {
    const props = {
        paramCategory : 1,
        category: {
            id: 1,
            name: 'cat1',
            categories: [{
                name: 'cat2',
                id: 2
            },
                {name: 'cat3', id: 3}]
        },
        onChange: (i) => { return i; },
        onClearSearch: function(){}
    };
    const tree = renderer
        .create(<Router><Vertical {...props} /></Router>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('test Vertical methods', function(){
    let props1 = {
        paramCategory : 2,
        category: {
            id: 1
        }
    };
    it('should return false if not selected category', function(){
        const wrapper = shallow(<Vertical {...props1} />);
        expect(wrapper.instance().isSelectedCategory()).toBeFalsy();
    });

    let props2 = {
        paramCategory : '1',
        category: {
            id: 1
        }
    };

    it('should return true if selected category', function(){
        const wrapper = shallow(<Vertical {...props2} />);
        expect(wrapper.instance().isSelectedCategory()).toBeTruthy();
    });
});