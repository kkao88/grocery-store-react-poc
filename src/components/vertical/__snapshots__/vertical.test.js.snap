// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical renders correctly 1`] = `
<div>
  <span>
    -&gt;
  </span>
  <a
    href="/category/1"
    onClick={[Function]}
  >
    <button
      className="btn"
    >
      cat1
    </button>
  </a>
  <ul>
    <li>
      <a
        href="/category/1/2"
        onClick={[Function]}
      >
        <button
          className="btn"
        >
          cat2
        </button>
      </a>
    </li>
    <li>
      <a
        href="/category/1/3"
        onClick={[Function]}
      >
        <button
          className="btn"
        >
          cat3
        </button>
      </a>
    </li>
  </ul>
</div>
`;
